### DISCUSSION TITLE
Welcome to Arabisc Discussions!

### DISCUSSION THREAD

**Discussion by ymoslem** [ymoslem/Arabisc#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what makes you interested in Arabisc.

**Comment by dragonnikkirocks** [comment]
Hi!!
Looks really interesting. Good Job! :)
How much is the processing time for a sentence using this approach?

**Reply by ymoslem** [reply]
@dragonnikkirocks Thanks for your message!

A sentence of 15 words takes about 3 seconds. In our implementation, we have 15 tokens/words as the maximum, and then you need to load more if needed. If you process fewer words, the time will be less (e.g. 7 words for about 2 seconds).

This is without loading the model and tokenizer, which should be handled separately so that we do not have to load them each time we check text in the current document. Other performance optimization approaches can be followed as well. For example, while writing this reply, the tokenizer is built each time the script is run; however, saving the tokenizer to JSON and then loading the JSON file only would save much time when loading the tokenizer.

The current script was mainly for testing the quality of our approach in terms of spelling correction suggestions and minimizing overcorrections, but I hope now you have an idea about the performance in terms of the processing time and required improvements.

Kind regards,
Yasmin
